telegram/ext/utils/promise.py

```diff
@@ -83,6 +84,11 @@ def run(self) -> None:
 
         finally:
             self.done.set()
+            if self._done_callback:
+                try:
+                    self._done_callback(self.result())
+                except Exception:
+                    pass
```

Comment:
When re-reading this, Noams force-ghost appeared and scolted me :D let's put a log here, something like
```suggestion
                    logger.warning('`done_callback` of a Promise raised the following exception. The exception won't be handled by error handlers.')
                    logger.warning(<traceback>)
```
that should also make codacy happy

Reply:
Lol, yea makes sense